lib/src/main/java/graphql/nadel/validation/NadelHydrationArgumentValidation.kt

```diff
@@ -0,0 +1,346 @@
+import graphql.nadel.validation.NadelSchemaValidationError.IncompatibleHydrationInputObjectArgumentType
+import graphql.nadel.validation.NadelSchemaValidationError.MissingHydrationArgumentValueSource
+import graphql.nadel.validation.NadelSchemaValidationError.MissingHydrationFieldValueSource
+import graphql.nadel.validation.NadelSchemaValidationError.MissingHydrationInputObjectArgumentValueSource
+import graphql.nadel.validation.NadelSchemaValidationError.MissingRequiredHydrationActorFieldArgument
+import graphql.nadel.validation.NadelSchemaValidationError.NonExistentHydrationActorFieldArgument
+import graphql.schema.GraphQLFieldDefinition
+import graphql.schema.GraphQLFieldsContainer
+import graphql.schema.GraphQLInputObjectType
+import graphql.schema.GraphQLNamedType
+import graphql.schema.GraphQLObjectType
+import graphql.schema.GraphQLScalarType
+import graphql.schema.GraphQLType
+
+internal class NadelHydrationArgumentValidation(
+    private val typeValidation: NadelTypeValidation,
+) {
+    fun validate(
+        parent: NadelServiceSchemaElement,
+        overallField: GraphQLFieldDefinition,
+        hydration: UnderlyingServiceHydration,
+        actorField: GraphQLFieldDefinition,
+    ): List<NadelSchemaValidationError> {
+        // Can only provide one value for an argument
+        val duplicatedArgumentsErrors = hydration.arguments
+            .groupBy { it.name }
+            .filterValues { it.size > 1 }
+            .values
+            .map {
+                DuplicatedHydrationArgument(parent, overallField, it)
+            }
+
+        val remoteArgErrors = mutableListOf<NadelSchemaValidationError>()
+        hydration.arguments.mapNotNull { remoteArg ->
+            val actorFieldArgument = actorField.getArgument(remoteArg.name)
+            if (actorFieldArgument == null) {
+                remoteArgErrors.add(NonExistentHydrationActorFieldArgument(
+                    parent,
+                    overallField,
+                    hydration,
+                    argument = remoteArg.name,
+                ))
+            } else {
+                remoteArgErrors.addAll(getRemoteArgErrors(parent, overallField, remoteArg, actorField))
+            }
+        }
+
+        val missingActorArgErrors = actorField.arguments
+            .filter { it.type.isNonNull }
+            .mapNotNull { actorArg ->
+                val hydrationArg = hydration.arguments.find { it.name == actorArg.name }
+                if (hydrationArg == null) {
+                    MissingRequiredHydrationActorFieldArgument(
+                        parent,
+                        overallField,
+                        hydration,
+                        argument = actorArg.name,
+                    )
+                } else {
+                    null
+                }
+            }
+
+        return duplicatedArgumentsErrors + remoteArgErrors + missingActorArgErrors
+    }
+
```

Comment:
everything above this moved from NadelHydrationArgumentValidation class, everything below is new